File: android/src/main/java/expo/modules/tutorialnative/MMKVStorage.kt
package expo.modules.tutorialnative

import android.content.Context
import com.tencent.mmkv.MMKV
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch

class MMKVStorage(context: Context) {
    private var mmkv: MMKV? = null

    init {
        CoroutineScope(Dispatchers.Main).launch {
            MMKV.initialize(context)
            mmkv = MMKV.defaultMMKV()
        }
    }


    fun get(key: String): String? {
        return mmkv?.getString(key, "")
    }

    fun set(key: String, value: String) {
        mmkv?.putString(key, value)
    }
}

File: ios/MMKVStorage.swift
import MMKV

class MMKVStorage {

  var mmkv: MMKV?

  init() {
    DispatchQueue.main.async {
      MMKV.initialize(rootDir: nil)
      self.mmkv = MMKV.default()
    }
  }


  func set(key: String, value: String) {
    mmkv?.set(value, forKey: key)
  }


  func get(key: String) -> String? {
    mmkv?.string(forKey: key, defaultValue: "")
  }

}
